src/mare/compiler/binary.cr

```diff
@@ -24,7 +24,14 @@ class Mare::Compiler::Binary
     machine = ctx.code_gen.target_machine
     mod = ctx.code_gen.mod
 
-    ponyrt_bc = LLVM::MemoryBuffer.from_file(PONYRT_BC_PATH)
+    target = ctx.options.target
```

Comment:
Instead of relying on someone explicitly specifying a target via `ctx.options.target`, I think it would be more robust here to use `Target.new(machine.triple)`, so that by default we will pick up whatever triple was detected automatically by LLVM.

This should let you compile on FreeBSD without specifying an explicit `--target` option to the CLI, so that one should only need to specify `--target` when cross-compiling.

```suggestion
    target = Target.new(machine.triple)
```